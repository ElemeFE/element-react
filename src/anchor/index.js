import AnchorSmooth from './AnchorSmooth';
import GoTop from './GoTop';

export {AnchorSmooth,GoTop};