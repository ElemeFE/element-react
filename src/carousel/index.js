import Carousel from './Carousel';
import CarouselItem from './CarouselItem';

Carousel.Item = CarouselItem;

export default Carousel;
