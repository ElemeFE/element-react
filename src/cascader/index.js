import Cascader from './Cascader';

export default Cascader;
