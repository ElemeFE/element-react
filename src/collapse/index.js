import Collapse from './Collapse';
import CollapseItem from './CollapseItem';

Collapse.Item = CollapseItem;

export default Collapse;
