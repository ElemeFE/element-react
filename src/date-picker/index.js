/*
revision version: 
  48996c6658300b7f9f01b9b4011a531f17e30b75
*/
export { default as TimeSelect } from './TimeSelect'
export { default as TimePicker } from './TimePicker'
export { default as TimeRangePicker } from './TimeRangePicker'
export { default as DatePicker } from './DatePicker'
export { default as DateRangePicker } from './DateRangePicker'