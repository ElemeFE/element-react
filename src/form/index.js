import Form from './Form';
import FormItem from './FormItem';

Form.Item = FormItem;

export default Form;
