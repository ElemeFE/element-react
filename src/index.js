export { default as i18n } from './locale';

export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Card } from './card';
export { default as Layout } from './layout';
export { default as Loading } from './loading';
export { default as Message } from './message';
export { default as MessageBox } from './message-box';
export { default as Notification } from './notification';
export { default as Radio } from './radio';
export { default as Dialog } from './dialog';
export { default as Rate } from './rate';
export { default as Progress } from './progress';
export { default as Badge } from './badge';
export { default as Tabs } from './tabs';
export { default as Tree } from './tree';
export { default as Input } from './input';
export { default as Icon } from './icon';
export { default as Menu } from './menu';
export { default as Steps } from './steps';
export { default as Breadcrumb } from './breadcrumb';
export { default as Tooltip } from './tooltip';
export { default as InputNumber } from './input-number';
export { default as Checkbox } from './checkbox';
export { default as Slider } from './slider';
export { default as Table } from './table'
export { default as Switch } from './switch';
export { default as Form } from './form';
export { default as Upload } from './upload';
export { default as Tag } from './tag';
export { default as Select } from './select';
export { default as Dropdown } from './dropdown';
export { default as Popover } from './popover';
export { default as Pagination } from './pagination';
export { default as AutoComplete } from './auto-complete';
export { TimeSelect, TimePicker, TimeRangePicker, DatePicker, DateRangePicker } from './date-picker';
export { default as Carousel } from './carousel';
export { default as Collapse } from './collapse';
export { default as ColorPicker } from './color-picker';
export { default as Cascader } from './cascader';
export { default as Transfer } from './transfer';
