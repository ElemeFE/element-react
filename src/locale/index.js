import format from './format';
import defaultLang from './lang/en';

let _lang = defaultLang

function use(lang) {
  _lang = lang;
}

function t(path, options) {
  const array = path.split('.');
    let current = _lang;

    for (var i = 0, j = array.length; i < j; i++) {
      var property = array[i];
      var value = current[property];
      if (i === j - 1) return format(value, options);
      if (!value) return '';
      current = value;
    }
    return '';
}

export default {
  use,
  t
}
