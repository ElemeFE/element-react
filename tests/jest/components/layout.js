import React from 'react';
import renderer from 'react-test-renderer';

import { Layout } from '../../../src';

test('Basic layout', () => {
  const component = renderer.create(
    <div>
      <Layout.Row>
        <Layout.Col span="24"><div className="grid-content bg-purple-dark"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row>
        <Layout.Col span="12"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="12"><div className="grid-content bg-purple-light"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row>
        <Layout.Col span="8"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="8"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="8"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple-light"></div></Layout.Col>
      </Layout.Row>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Column spacing', () => {
  const component = renderer.create(
    <Layout.Row gutter="20">
      <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
    </Layout.Row>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Hybrid layout', () => {
  const component = renderer.create(
    <div>
      <Layout.Row gutter="20">
        <Layout.Col span="16"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="8"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row gutter="20">
        <Layout.Col span="8"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="8"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row gutter="20">
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="16"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="4"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Column offset', () => {
  const component = renderer.create(
    <div>
      <Layout.Row gutter="20">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6" offset="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row gutter="20">
        <Layout.Col span="6" offset="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6" offset="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row gutter="20">
        <Layout.Col span="12" offset="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Alignment', () => {
  const component = renderer.create(
    <div>
      <Layout.Row type="flex" className="row-bg">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row type="flex" className="row-bg" justify="center">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row type="flex" className="row-bg" justify="end">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row type="flex" className="row-bg" justify="space-between">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
      <Layout.Row type="flex" className="row-bg" justify="space-around">
        <Layout.Col span="6"><div className="grid-content bg-purple"></div></Layout.Col>
        <Layout.Col span="6"><div className="grid-content bg-purple-light"></div></Layout.Col>
        <Layout.Col span={6}><div className="grid-content bg-purple"></div></Layout.Col>
      </Layout.Row>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Responsive Layout', () => {
  const component = renderer.create(
    <Layout.Row gutter="10">
      <Layout.Col xs="8" sm="6" md="4" lg="3"><div className="grid-content bg-purple"></div></Layout.Col>
      <Layout.Col xs="4" sm="6" md="8" lg="9"><div className="grid-content bg-purple-light"></div></Layout.Col>
      <Layout.Col xs="4" sm="6" md="8" lg="9"><div className="grid-content bg-purple"></div></Layout.Col>
      <Layout.Col xs="8" sm="6" md="4" lg="3"><div className="grid-content bg-purple-light"></div></Layout.Col>
    </Layout.Row>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
