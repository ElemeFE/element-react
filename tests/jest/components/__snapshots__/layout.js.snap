// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment 1`] = `
<div>
  <div
    className="el-row el-row--flex row-bg"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row is-justify-center el-row--flex row-bg"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row is-justify-end el-row--flex row-bg"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row is-justify-space-between el-row--flex row-bg"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row is-justify-space-around el-row--flex row-bg"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
</div>
`;

exports[`Basic layout 1`] = `
<div>
  <div
    className="el-row"
    style={Object {}}
  >
    <div
      className="el-col el-col-24"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-dark"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={Object {}}
  >
    <div
      className="el-col el-col-12"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-12"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={Object {}}
  >
    <div
      className="el-col el-col-8"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-8"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-8"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={Object {}}
  >
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={Object {}}
  >
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={Object {}}
    >
      <div
        className="grid-content bg-purple-light"
      />
    </div>
  </div>
</div>
`;

exports[`Column offset 1`] = `
<div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-6"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6 el-col-offset-6"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-6 el-col-offset-6"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-6 el-col-offset-6"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-12 el-col-offset-6"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
</div>
`;

exports[`Column spacing 1`] = `
<div
  className="el-row"
  style={
    Object {
      "marginLeft": "-10px",
      "marginRight": "-10px",
    }
  }
>
  <div
    className="el-col el-col-6"
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
  <div
    className="el-col el-col-6"
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
  <div
    className="el-col el-col-6"
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
  <div
    className="el-col el-col-6"
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
</div>
`;

exports[`Hybrid layout 1`] = `
<div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-16"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-8"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-8"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-8"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
  <div
    className="el-row"
    style={
      Object {
        "marginLeft": "-10px",
        "marginRight": "-10px",
      }
    }
  >
    <div
      className="el-col el-col-4"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-16"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
    <div
      className="el-col el-col-4"
      style={
        Object {
          "paddingLeft": "10px",
          "paddingRight": "10px",
        }
      }
    >
      <div
        className="grid-content bg-purple"
      />
    </div>
  </div>
</div>
`;

exports[`Responsive Layout 1`] = `
<div
  className="el-row"
  style={
    Object {
      "marginLeft": "-5px",
      "marginRight": "-5px",
    }
  }
>
  <div
    className="el-col el-col-24 el-col-xs-8 el-col-sm-6 el-col-md-4 el-col-lg-3"
    style={
      Object {
        "paddingLeft": "5px",
        "paddingRight": "5px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
  <div
    className="el-col el-col-24 el-col-xs-4 el-col-sm-6 el-col-md-8 el-col-lg-9"
    style={
      Object {
        "paddingLeft": "5px",
        "paddingRight": "5px",
      }
    }
  >
    <div
      className="grid-content bg-purple-light"
    />
  </div>
  <div
    className="el-col el-col-24 el-col-xs-4 el-col-sm-6 el-col-md-8 el-col-lg-9"
    style={
      Object {
        "paddingLeft": "5px",
        "paddingRight": "5px",
      }
    }
  >
    <div
      className="grid-content bg-purple"
    />
  </div>
  <div
    className="el-col el-col-24 el-col-xs-8 el-col-sm-6 el-col-md-4 el-col-lg-3"
    style={
      Object {
        "paddingLeft": "5px",
        "paddingRight": "5px",
      }
    }
  >
    <div
      className="grid-content bg-purple-light"
    />
  </div>
</div>
`;
